tools/codegen/src/generator/languages/rust/utilities.rs

```diff
@@ -1,15 +1,97 @@
 use case::CaseExt;
 use proc_macro2 as m4;
 
+use std::{collections::HashSet, sync::OnceLock};
+
+static RUST_KEYWORDS: OnceLock<HashSet<String>> = OnceLock::new();
+
+fn rust_keyword() -> &'static HashSet<String> {
+    RUST_KEYWORDS.get_or_init(|| {
+        IntoIterator::into_iter([
+            // strick
+            "as",
+            "struct",
+            "break",
+            "const",
+            "continue",
+            "crate",
+            "else",
+            "enum",
+            "extern",
+            "false",
+            "fn",
+            "for",
+            "if",
+            "impl",
+            "in",
+            "let",
+            "loop",
+            "match",
+            "mod",
+            "move",
+            "mut",
+            "pub",
+            "ref",
+            "return",
+            "self",
+            "Self",
+            "static",
+            "super",
+            "trait",
+            "true",
+            "type",
+            "unsafe",
+            "use",
+            "where",
+            "while",
+            // 2018+
+            "async",
+            "await",
+            "dyn",
+            // reserved
+            "abstract",
+            "become",
+            "box",
+            "do",
+            "final",
+            "macro",
+            "override",
+            "priv",
+            "typeof",
+            "unsized",
+            "virtual",
+            "yield",
+            // 2018+
+            "try",
+            // weak
+            "macro_rules",
+            "union",
+            "'static",
+        ])
+        .map(ToString::to_string)
+        .collect()
+    })
+}
+
+fn avoid_keywords(name: &str) -> String {
+    let keywords = rust_keyword();
+    let snake_name = name.to_snake();
+    if keywords.contains(&snake_name) {
+        snake_name + "_"
+    } else {
+        snake_name
+    }
+}
+
```

Comment:
If the user inputs `Struct` as a field name, the function converts it to lowercase and appends an underscore, changing it to `struct_`.

Perhaps it doesn't need to be converted to lowercase; appending `_` is sufficient, so it should just become `Struct_`?

Reply:
Then the function name `avoid_keywords` does not make sense.

Reply:
I asked GPT-4, and it suggested renaming it to `sanitize_identifier`.